testsuite/smoke-test/src/aptos_cli.rs

```diff
@@ -140,16 +163,100 @@ async fn test_register_and_update_validator() {
         .await
         .unwrap();
     assert_eq!(
-        bls12381::PublicKey::from_encoded_string(
-            validator_config
-                .as_object()
-                .unwrap()
-                .get("consensus_pubkey")
-                .unwrap()
-                .as_str()
-                .unwrap()
-        )
-        .unwrap(),
-        consensus_public_key
+        validator_config.consensus_public_key,
+        keys.consensus_public_key()
+    );
+
+    let new_port = 5678;
+    let new_network_private_key = keygen.generate_x25519_private_key().unwrap();
+
+    cli.update_validator_network_addresses(
+        validator_cli_index,
+        HostAndPort {
+            host: dns_name("0.0.0.0"),
+            port: new_port,
+        },
+        new_network_private_key.public_key(),
+    )
+    .await
+    .unwrap();
+
+    tokio::time::sleep(Duration::from_secs(2)).await;
```

Comment:
Is this for waiting long enough so the tx is executed? 2 secs seems like a magic number

Reply:
We should probably work on a state based check vs just a raw sleep

Reply:
this was unnecessary, left from previous testing. CliTestFramework wait for transactions results, and returns transactions, so should be good.